Engine/src/Console/CVar.bf

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections;
+
+namespace SteelEngine.Console
+{
+	abstract class CVar
+	{
+		protected String _name ~ delete _;
+		protected String _help ~ delete _;
+
+		public StringView Name => _name;
+		public StringView Help => _help;
+		public CVarFlags Flags { get; protected set; }
+
+		public bool HasFlags(CVarFlags flags) => Flags.HasFlag(flags); 
+		public void AddFlags(CVarFlags flags) => Flags |= flags;
+		public void RemoveFlags(CVarFlags flags) => Flags &= ~flags;
+
+		public abstract Type Type { get; }
+
+		public abstract int32 GetValueInt32();
+		public abstract int64 GetValueInt64();
+		public abstract float GetValueFloat() ;
+		public abstract StringView GetValueString(String buffer) ;
+
+		public abstract Result<bool> Execute(StringView strArgs, Span<StringView> args);
+
+		public virtual bool IsCommand => false;
+
+		protected this(StringView name, StringView help, CVarFlags flags)
+		{
+			_name = new .(name);
+			_help = new .(help);
+			Flags = flags;
+		}
+	}
+
+	public delegate void OnCVarChange(CVar cvar);
+
+	class ConsoleVar<T> : CVar where T : var
+	{
+		protected T* _value;
+
+		public ref T Value => *_value;
```

Comment:
This seems a bit dangerous if the value is null. We should have null protections or require that a CVar be non-null